src/sqlparser/src/parser.rs

```diff
@@ -2612,6 +2623,29 @@ impl Parser<'_> {
             None
         };
 
+        let webhook_info = if self.parse_keyword(Keyword::VALIDATE) {
+            if !contain_webhook {
+                parser_err!("VALIDATE is only supported for tables created with webhook source");
+            }
+            self.expect_keyword(Keyword::SECRET)?;
+            let secret_ref = SecretRef {
+                secret_name: self.parse_object_name()?,
+                ref_as: SecretRefAsType::Text,
+            };
+            if self.parse_keywords(&[Keyword::AS, Keyword::FILE]) {
+                parser_err!("Secret for SECURE_COMPARE() does not support AS FILE");
+            };
```

Comment:
Shall we extract the snippet for parsing secret ref?

Reply:
done